src/protocol/sbus_uart.c

```diff
@@ -84,12 +94,33 @@ static void build_rcdata_pkt()
 
 // static u8 testrxframe[] = { 0x00, 0x0C, 0x14, 0x00, 0x00, 0x00, 0x00, 0xFF, 0x01, 0x03, 0x00, 0x00, 0x00, 0xF4 };
 
+static enum {
+    ST_DATA1,
+    ST_DATA2,
+} state;
+
+static volatile int mixer_sync;
+static u16 mixer_runtime;
+static u16 sbus_period;
 static u16 serial_cb()
 {
-    build_rcdata_pkt();
-    UART_Send(packet, sizeof packet);
-
-    return SBUS_FRAME_PERIOD;
+    if (sbus_period != Model.proto_opts[PROTO_OPTS_PERIOD] * 1000)
```

Comment:
Well I hadn't noticed SETOPTIONS.  Seems like it opens a synchronization issue.  The options are used within an ISR so updating through SETOPTIONS opens the possibility of using a partially updated value.

Reply:
It is not a problem to atomic read/write. check this:
https://stackoverflow.com/questions/8107108/stm32-programming-tips-and-questions

Question 2: Sharing variables between interrupts and the main loop

People have written entire chapters explaining the answers to this group of questions. Whenever you share a variable between the main loop and an interrupt, you should definitely use the volatile keywords on it. Variables of 32 or fewer bits can be accessed atomically (unless they are misaligned).

If you need to access a larger variable, or two variables at the same time from the main loop, then you will have to disable the clock interrupt while you are accessing the variables. If your interrupt does not require precise timing, this will not be a problem. When you re-enable the interrupt, it will automatically fire if it needs to.

Reply:
The protocol options are 16bit in the Model struct and no guarantee they are aligned (though I guess we could force that).  Seems like that's better suited to a separate PR, especially since most protocols the same technique as here so they could all be cleaned up.

Reply:
sounds good to me. actually, i believe protocol code need some cleanup after we branch out for a new release. I am not happy with the fact all protocol has its own statemachine. We should consolidate the statemachine. Your change on mixer can be also implement in one single place.